### PR TITLE
add end to end test for liquid-vault

### PR DESCRIPTION
**Pull request by VitalR**
Please review the current branch with end to end test for liquid vault:
'should be able to claim 1 batch after 1 purchase with 0% fees and remove liquidity from uniswap'.

Thanks in advance.

### CODE REVIEW

File: test/liquid-vault.js [hunk 1]
```diff
@@ -337,5 +337,58 @@ describe('LiquidVault', function () {
...
+    const lpBalanceAfterClaim = await uniswapPair.balanceOf(owner.address);
+
+    assert.strictEqual(holder, owner.address);
+    assert.isTrue(claimed);
+    assertBNequal(amount, lockedLP[1]);
+    assertBNequal(exitFee, estimatedFeeAmount);
+    assertBNequal(amount.sub(exitFee), lpBalanceAfterClaim.sub(lpBalanceBeforeClaim));
+
+    infinityBalanceBefore = await infinity.balanceOf(accounts[0].address)
+    wethBalanceBefore = await weth.balanceOf(accounts[0].address)
```
Comment (ytumanov) [inline test/liquid-vault.js:368]: accounts[0].address --> owner.address

**Review by ytumanov — CHANGES_REQUESTED** [review]

File: test/liquid-vault.js (hunk 1 above)
Comment (VitalR) [reply test/liquid-vault.js]: resolved

File: test/liquid-vault.js [hunk 2]
```diff
@@ -378,16 +378,88 @@ describe('LiquidVault', function () {
...
+    assertBNequal(amount, lockedLP[1]);
+    assertBNequal(exitFee, estimatedFeeAmount);
+    assertBNequal(amount.sub(exitFee), lpBalanceAfterClaim.sub(lpBalanceBeforeClaim));
+
+    uniswapPairUserBalanceBefore = await uniswapPair.balanceOf(user.address);
+    uniswapPairOwnerBalanceBefore = await uniswapPair.balanceOf(owner.address);
+
+    await uniswapPair.transfer(user.address, lpBalanceAfterClaim);
+
+    uniswapPairUserBalanceAfter = await uniswapPair.balanceOf(user.address);
```
Comment (ytumanov) [inline test/liquid-vault.js:424]: lines 419-410, 424-425.  It's better to hardcode LP balance values in test to be able to compare hardcoded LP balances for user, owner before transfer on line 422 and after

File: test/liquid-vault.js [hunk 3]
```diff
@@ -378,16 +378,88 @@ describe('LiquidVault', function () {
...
+      new Date().getTime() + 3000,
+    )).to.emit(uniswapPair, 'Burn');
+
+    const infinityBalanceAfter = await infinity.balanceOf(owner.address);
+    const ethUserBalanceAfter = await ethers.provider.getBalance(user.address);
+    
+    assert.isTrue(infinityBalanceAfter.gt(infinityBalanceBefore));
+    assert.isTrue(ethUserBalanceAfter.gt(ethUserBalanceBefore));
+
+    const lpBalanceAfterRemoveLiquidity = await uniswapPair.balanceOf(owner.address);
```
Comment (ytumanov) [inline test/liquid-vault.js:460]: lpBalanceAfterRemoveLiquidity is zero from line 422. so You need to check that USER and not owner has LP tokens before removeLiquidityETH and doesn't have after transaction

**Review by ytumanov — CHANGES_REQUESTED** [review]

File: test/liquid-vault.js (hunk 2 above)
Comment (VitalR) [reply test/liquid-vault.js]: resolved

File: test/liquid-vault.js (hunk 3 above)
Comment (VitalR) [reply test/liquid-vault.js]: resolved

File: test/liquid-vault.js [hunk 4]
```diff
@@ -337,5 +337,124 @@ describe('LiquidVault', function () {
...
+      lpBalanceAfterClaim,
+      0,
+      0,
+      owner.address,
+      new Date().getTime() + 3000,
+    )).to.be.revertedWith('revert ds-math-sub-underflow');
+
+    const ethUserBalanceBeforeRemoveLiquidity = await ethers.provider.getBalance(user.address);
+
+    assert.notEqual(await uniswapPair.balanceOf(user.address), 0);
```
Comment (ytumanov) [inline test/liquid-vault.js:441]: Not needed

**Review by ytumanov — APPROVED** [review]

File: test/liquid-vault.js (hunk 4 above)
Comment (VitalR) [reply test/liquid-vault.js]: resolved

**Pull request merged**